packages/itwinui-react/src/core/ComboBox/ComboBoxMenu.tsx

```diff
@@ -49,19 +53,50 @@ const VirtualizedComboBoxMenu = (props: React.ComponentProps<'div'>) => {
     );
   }, [focusedIndex, menuRef]);
 
-  const { outerProps, innerProps, visibleChildren } = useVirtualization({
+  const virtualizer = useVirtualScroll({
     // 'Fool' VirtualScroll by passing length 1
     // whenever there is no elements, to show empty state message
-    itemsLength: filteredOptions.length || 1,
-    itemRenderer: virtualItemRenderer,
+    count: filteredOptions.length || 1,
+    getScrollElement: () => menuRef.current,
+    estimateSize: () => (mostlySubLabeled ? 48 : 36),
     scrollToIndex: focusedVisibleIndex,
+    gap: -1,
```

Comment:
There is a margin of `-1px` that is intended to exist between options on the `ComboBox`, but because the tanstack virtualizer relies on absolute positioning, setting the margin doesn't change the spacing between the items. Setting the `gap` to `-1` fixes this issue by having the virutalizer remove a pixel of space between each item, thus maintaining our original spacing.

Reply:
More discussion in [this thread](https://github.com/iTwin/iTwinUI/pull/2061#discussion_r1612062062).